frontend/src/metabase/actions/components/ActionViz/Action.tsx

```diff
@@ -114,16 +118,29 @@ const ConnectedActionComponent = connect()(ActionComponent);
 function mapStateToProps(state: State, props: ActionProps) {
   return {
     isEditingDashcard: getEditingDashcardId(state) === props.dashcard.id,
+    metadata: getMetadata(state),
   };
 }
 
 export function ActionFn(props: ActionProps) {
-  if (!props.dashcard?.action) {
+  const {
+    metadata,
+    dashcard: { action },
+  } = props;
+  const actionsEnabled = !!metadata
+    ?.database(action?.database_id)
+    ?.hasActionsEnabled?.();
+
+  if (!props.dashcard?.action || !actionsEnabled) {
```

Comment:
I believe we can just do `if (!action || !actionsEnabled)` here since `props` and `dashcard` are already destructured?